gprofiler/profilers/perf.py

```diff
@@ -202,11 +210,50 @@ def snapshot(self) -> ProcessToProfileData:
         for perf in self._perfs:
             perf.switch_output()
 
-        return {
-            # TODO generate appids for non runtime-profiler processes here
-            k: ProfileData(v, None, None)
-            for k, v in merge.merge_global_perfs(
-                self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
-                self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
-            ).items()
-        }
+        profile_data_dict = {}
+
+        for k, v in merge.merge_global_perfs(
+            self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
+            self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
+        ).items():
+            metadata = None
+            try:
+                if self._golang_metadata.is_golang_process(Process(k)):
+                    metadata = self._golang_metadata.make_application_metadata(Process(k))
+            except (NoSuchProcess, FileNotFoundError):
+                pass
+            profile_data_dict[k] = ProfileData(v, None, metadata)
+
+        return profile_data_dict
+
+
+class GolangMetadata(ApplicationMetadata):
+    def is_golang_process(self, process: Process) -> bool:
+        version = self._get_golang_version(process)
+        return version is not None
+
+    @functools.lru_cache(maxsize=128)
+    def _get_golang_version(self, process: Process) -> Optional[str]:
+        elf_path = f"/proc/{process.pid}/exe"
+        symbol_data = read_elf_symbol(elf_path, "runtime.buildVersion", 16)
+        if symbol_data is None:
+            return None
+        # Declaration of go string type:
+        # type stringStruct struct {
+        # 	str unsafe.Pointer
+        # 	len int
+        # }
+        addr, length = struct.unpack("<QQ", symbol_data)
```

Comment:
```suggestion
        addr, length = struct.unpack("QQ", symbol_data)
```

gProfiler also runs on ARM, IDK if it's LE or BE